connectors/flink/src/main/java/io/delta/flink/source/internal/enumerator/supplier/KernelSnapshotWrapper.java

```diff
@@ -0,0 +1,154 @@
+package io.delta.flink.source.internal.enumerator.supplier;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.standalone.DeltaScan;
+import io.delta.standalone.actions.AddFile;
+import io.delta.standalone.actions.Metadata;
+import io.delta.standalone.data.CloseableIterator;
+import io.delta.standalone.data.RowRecord;
+import io.delta.standalone.expressions.Expression;
+
+
+/**
+ * Wrap a {@link io.delta.kernel.Snapshot} such that it conforms to the the {@link
+ * io.delta.standalone.Snapshot} interface.
+ *
+ * NB: Currently only supports the getMetadata and getVersion methods. All others throw exceptions.
+ */
+
+public class KernelSnapshotWrapper implements io.delta.standalone.Snapshot {
```

Comment:
Since eventually standalone snapshot will be removed... would it make cleaner to have a common parent trait , say DeltaSnapshot, and 2 simple wrappers - StandaloneSnapshot and KernelSnapshot? Then it will be less code change to remove Standalone completely as we simple remove the trait and StandaloneSnapshot wrapper.

Reply:
Also, Standalone Snapshot class is not really meant to be extended like this. So i suggest we do it the wrapper way like this. This can become the model of any connector migration that we will recommend to the world.